src/gui/shutdownconfirm.cpp

```diff
@@ -33,50 +33,81 @@
 #include "base/types.h"
 #include "shutdownconfirm.h"
 
+#include <QVBoxLayout>
+#include <QHBoxLayout>
+#include <QStyle>
+#include <QIcon>
+#include <QLabel>
+#include <QDialogButtonBox>
+#include <QCheckBox>
 #include <QPushButton>
 
 ShutdownConfirmDlg::ShutdownConfirmDlg(const ShutdownAction &action)
-    : m_exitNow(0)
+    : m_neverShowAgain(false)
     , m_timeout(15)
     , m_action(action)
 {
-    // Title and button
+    QVBoxLayout *myLayout = new QVBoxLayout();
+    //Warning Icon and Text
+    QHBoxLayout *messageRow = new QHBoxLayout();
+    QLabel *warningLabel = new QLabel();
+    QIcon warningIcon(style()->standardIcon(QStyle::SP_MessageBoxWarning, 0, this));
+    warningLabel->setPixmap(warningIcon.pixmap(warningIcon.actualSize(QSize(32, 32))));
+    messageRow->addWidget(warningLabel);
+    m_text = new QLabel();
+    messageRow->addWidget(m_text);
+    myLayout->addLayout(messageRow);
```

Comment:
>I vote for making most of these variables private members of the class.

Don't need to do this if access to it is not required anywhere else.
>Also you allocate memory here but you don't delete it in the destructor.

It should to pass `parent` parameter to widget constructors then they'll be deleted in parent destructor.